docs/topics/migrations.txt

```diff
@@ -276,6 +276,14 @@ class to make it importable::
 Please refer to the notes about :ref:`historical-models` in migrations to see
 the implications that come along.
 
+.. _initial-property:
+
+.. note::
+
+    You may add an ``initial = True`` class property to the ``Migration`` class
+    to have Django apply the "initial migration" logic when the
```

Comment:
I don't think this section will be very long regardless. This is the canonical location for explaining this behavior, so I think it's more important that we explain it clearly than that we keep it short. Here's my proposal:

> The "initial migrations" for an app are the migrations that create the first version of that app's tables. Usually an app will have just one initial migration, but in some cases of complex model interdependencies it may have two or more.
>
> Initial migrations are marked with an `initial = True` class attribute on the migration class. If no `initial` class attribute is found, a migration will be considered "initial" if it is the first migration in the app (i.e. it has no dependencies on any other migration in the same app).
>
> When :djadmin:`migrate` is run with the `--fake-initial` argument, these initial migrations are treated specially. For an initial migration that creates a one or more tables (`CreateModel` operation), Django will check if those tables already exist in the database, and fake-apply the migration if so. Similarly, for an initial migration that adds one or more fields (`AddField` operation), Django will check if those columns already exist in the database, and fake-apply the migration if so. When :djadminopt:`--fake-initial` is not used, initial migrations are treated no differently from any other migration.
>
> When auto-generating migrations, Django should always set the `initial` flag correctly. Modifying the flag manually is for advanced use cases only; if you aren't sure whether you need it, you probably don't.